### DISCUSSION TITLE
Fork for CYD (Cheap-Yellow-Display)

### DISCUSSION THREAD

**Discussion by Lukaswnd** [esp-cpp/esp-box-emu#63] [Q&A]
Hey, has anyone tried to make this lib work with one of the CYD?
In generall I think It would be a great Idea to design this a little more modular. I know this repo is supossed to fokus on the ESP BOX. But it became one of the best emulators for the esp and one which is still maintained. So why not make it for everyone? I think the ESP Box is not that cheep/ there are cheeper options. And a lot of people have an esp, a display and some buttons in their inventory. By making the Display, the input, the audio codec and the sdcard a little more modular, one could disable the default Classes for the ESP Box and include their own implementation for their hardware. In general I don't think it is hard to make these changes, they just take some time to refactor a little. The only problem I see are Displays with a different resolution. I'd recommend to forid smaller Displays and make only use a part of larger screens. more could be done later.

Unfortunaly I don't know much about the implementation of this library and have not enough time to learn it. And I think a lot of you are interested in the finished produkt: Buy the Box, add Inputboard and print a case to get your emulator. 
So are there other people interested in more modularity? And how much efford would it really be, I just scanned a view files and forks?

**Comment by finger563** [comment]
@Lukaswnd Thanks for starting this discussion and your interest in the project :)

Regarding your question, that's actually on my to-do list to make this project easier to customize for different hardware. I think a combination of `menuconfig` settings for things like `sdcard`, io configuration, screen driver, audio driver, etc. as well as some refactoring of the component subsystems to make them easier to swap out for custom implementations would indeed go a long way. Definitely happy to discuss this further and I think a good first approach would be selecting another alternate hardware config to evaluate the changes needed to support it and hopefully make those changes in a way that more easily support more diverse hardware down the line.

**Reply by finger563** [reply]
@Lukaswnd I know this is a relatively stale discussion, but I wanted to highlight some renewed interest in supporting additional hardware. Please see the discussion over at #95 if you're still interested.
